benchmarks/workloads.py

```diff
@@ -133,3 +137,54 @@ def large_dense_attr_hex(cls, x: int = 1024, y: int = 1024) -> str:
             )
         ]
         return WorkloadBuilder.wrap_module(ops)
+
+    @classmethod
+    def large_constant_tensor(
+        cls,
+        tensor_shape: tuple[int, ...],
+        num_add_ops: int = 10,
+    ) -> ModuleOp:
+        """Create a module with a function that adds multiple large constant tensors.
+
+        Args:
+            tensor_shape: The shape of the constant tensors to be created.
+            num_add_ops: The number of addition operations to perform with the constant tensors.
+
+        Returns:
+            The created module containing the function with addition operations.
+        """
+        tensor_type = TensorType(shape=tensor_shape, element_type=Float32Type())
+        function_type = FunctionType.from_lists(
+            inputs=[tensor_type], outputs=[tensor_type]
+        )
+
+        func_op = FuncOp(name="main", function_type=function_type)
+
+        input_val = func_op.args[0]
+        output_val = input_val
+
+        for _ in range(num_add_ops):
+            # Create a constant tensor with random values
+            dense_elements_attr = DenseIntOrFPElementsAttr.from_list(
+                type=tensor_type,
+                data=[random.random() for _ in range(prod(tensor_shape))],
```

Comment:
having randomness in tests feels like a bad idea to me, if it really needs to "look" random, can we at least fix a seed?

Reply:
Or floats from 0 to size.

Reply:
It doesn't particularly need to be random -- this was just lifted from the example code. However, I don't really see any difference between a fixed set of random numbers generated from a seed and an arbitrary selection made during development, other than perhaps being slightly more representative as Python might be able to do strange optimisation trickery for allocating big homogeneous lists?

We could change it to this if you particularly want to?

```suggestion
                data=[0.0 for _ in range(prod(tensor_shape))],
```

Reply:
Is the time taken to generate this list part of the benchmark time? (Basically does the time taken to generate random numbers matter?) If not I would be happy with the random numbers